internal/patchpkg/builder.go

```diff
@@ -302,6 +322,65 @@ func (d *DerivationBuilder) findRemovedRefs(ctx context.Context, pkg *packageFS)
 	return refs, nil
 }
 
+func (d *DerivationBuilder) findCUDA(ctx context.Context, out *packageFS) error {
```

Comment:
Food for future thought: Is there a way to do this in a more plugin-like way? Maybe we allow built in plugins to use go code (or in the future compile plugins as needed).

Reply:
I like the idea of aligning plugins and patching. This code runs as part of the flake builder (in the nix build sandbox), so plugins would just need something similar.